packages/network-controller/src/provider-api-tests/shared-tests.ts

```diff
@@ -0,0 +1,2673 @@
+          response: { result: mockResults[0] },
+        });
+
+        comms.mockRpcCall({
+          request: requests[1],
+          response: { result: mockResults[1] },
+        });
+
+        const results = await withNetworkClient(
+          { providerType },
+          ({ makeRpcCallsInSeries }) => makeRpcCallsInSeries(requests),
+        );
+
+        expect(results).toStrictEqual([mockResults[0], mockResults[1]]);
+      });
+    });
+  }
+};
+
+type TestsForRpcMethodSupportingBlockParam = {
+  blockParamIndex: number;
+  numberOfParameters: number;
+  providerType: 'infura' | 'custom';
+};
+
+export const testsForRpcMethodSupportingBlockParam = (
+  method: string,
+  {
+    blockParamIndex,
+    numberOfParameters,
+    providerType,
+  }: TestsForRpcMethodSupportingBlockParam,
+) => {
+  describe.each([
+    ['given no block tag', undefined],
+    ['given a block tag of "latest"', 'latest'],
+  ])('%s', (_desc, blockParam) => {
+    it('does not hit the RPC endpoint more than once for identical requests', async () => {
+      const requests = [
+        {
+          method,
+          params: buildMockParams({ blockParamIndex, blockParam }),
+        },
+        { method, params: buildMockParams({ blockParamIndex, blockParam }) },
+      ];
+      const mockResults = ['first result', 'second result'];
+
+      await withMockedCommunications({ providerType }, async (comms) => {
+        // The first time a block-cacheable request is made, the block-cache
+        // middleware will request the latest block number through the block
+        // tracker to determine the cache key. Later, the block-ref
+        // middleware will request the latest block number again to resolve
+        // the value of "latest", but the block number is cached once made,
+        // so we only need to mock the request once.
+        comms.mockNextBlockTrackerRequest({ blockNumber: '0x100' });
+        // The block-ref middleware will make the request as specified
+        // except that the block param is replaced with the latest block
+        // number.
+        comms.mockRpcCall({
+          request: buildRequestWithReplacedBlockParam(
+            requests[0],
+            blockParamIndex,
+            blockParam === undefined ? null : 'latest',
+          ),
+          response: { result: mockResults[0] },
+        });
+
+        const results = await withNetworkClient(
+          { providerType },
+          ({ makeRpcCallsInSeries }) => makeRpcCallsInSeries(requests),
+        );
+
+        expect(results).toStrictEqual([mockResults[0], mockResults[0]]);
+      });
+    });
+
+    for (const paramIndex of [...Array(numberOfParameters).keys()]) {
+      if (paramIndex === blockParamIndex) {
+        // testing changes in block param is covered under later tests
+        continue;
+      }
+
+      it(`does not reuse the result of a previous request if parameter at index "${paramIndex}" differs`, async () => {// eslint-disable-line
```

Comment:
What was the lint violation here?

Reply:
If I remember correctly, it was yelling about defining a function inside a loop.